packages/eslint-plugin/src/rules/prefer-as-const.ts

```diff
@@ -0,0 +1,67 @@
+import {
+  AST_NODE_TYPES,
+  TSESTree,
+} from '@typescript-eslint/experimental-utils';
+import * as util from '../util';
+
+export default util.createRule({
+  name: 'prefer-as-const',
+  meta: {
+    type: 'suggestion',
+    docs: {
+      description: 'Prefer usage of `as const` over literal type',
+      category: 'Best Practices',
+      recommended: false,
+    },
+    fixable: 'code',
+    messages: {
+      preferConstAssertion:
+        'Expected a `const` instead of a literal type assertion',
+      variableConstAssertion:
+        'Expected a `const` assertion instead of a literal type annotation',
+    },
+    schema: [],
+  },
+  defaultOptions: [],
+  create(context) {
+    function compareTypes(
+      valueNode: TSESTree.Expression,
+      typeNode: TSESTree.TypeNode,
+      canFix: boolean,
+    ): void {
+      if (
+        valueNode.type === AST_NODE_TYPES.Literal &&
+        typeNode.type === AST_NODE_TYPES.TSLiteralType &&
+        'raw' in typeNode.literal &&
+        valueNode.raw === typeNode.literal.raw
+      ) {
+        if (canFix) {
+          context.report({
+            node: typeNode,
+            messageId: 'preferConstAssertion',
+            fix: fixer => fixer.replaceText(typeNode, 'const'),
+          });
+        } else {
+          context.report({
+            node: typeNode,
+            messageId: 'variableConstAssertion',
+          });
```

Comment:
nit - could use a suggestion in this case instead. Suggestions aren't applied automatically, and have to be applied via an IDE, so it allows you to have dangerous and "breaking" fixers.

```ts
          context.report({
            node: typeNode,
            messageId: 'variableConstAssertion',
            suggestions: [
              {
                messageId: 'someId',
                fix: fixer => { ... },
              },
            ],
          });
```

`no-explicit-any` and `no-non-null-assertion` both have fixers (same as #1469)

Reply:
i'm not familiar with suggestion API, but from what i see its just working same way as fixer but its optional?